website/src/app/kb/automate/terraform/azure/readme.mdx

```diff
@@ -0,0 +1,117 @@
+   if you haven't yet done so.
+1. A [Firezone Site](https://www.firezone.dev/kb/deploy/sites) dedicated to use
+   for this example. This Site should contain **only** the Firezone Gateway(s)
+   deployed in this example and any associated Resources.
+1. A Firezone Gateway token. This can be obtained by viewing your Site in the
+   admin portal, clicking the `Deploy Gateway` button, and navigating to the
+   instructions for the `Azure` tab. Gateway tokens support multi-use, so only a
+   single token is needed to provision the Firezone Gateways in this guide.
+
+## Sizing
+
+Simply update the number of `desired_capacity` to deploy more or fewer Firezone
+Gateways. There's no limit to the number of Firezone Gateways you can deploy in
+a single Vnet. A basic Azure Autoscale configuration is provisioned as part of
+the linked module.
+
+We've tested with `Standard_B1ls` instances which still work quite well for most
+applications. However, you may want to consider a larger instance type if you
+have a high volume of traffic or lots of concurrent connections.
+
+## Deployment
+
+1. [Download](https://raw.githubusercontent.com/firezone/firezone/main/terraform/examples/azure/nat-gateway/main.tf)
+   the `main.tf` from the example module.
+1. Customize it as desired. At a minimum, you will need to set the
+   `firezone_token` and change `admin_ssh_key` to match your environment.
+1. Run `terraform init` to initialize the working directory and download the
+   required providers.
+1. Run `terraform apply` to deploy the Firezone Gateway(s) into your AWS
+   project.
+
+You can see the IP address assigned to the NAT Gateway in the Terraform output.
+These are the IP address that your Firezone Gateway(s) will share to egress
+traffic.
+
+<Image
+  src="/images/kb/automate/terraform/azure/gateways.png"
+  alt="Firezone Gateways in the Azure portal"
+  width={800}
+  height={600}
+/>
+
+<Alert color="info">
+  It can take a few minutes for the Firezone Gateway(s) to provision and connect
+  to the portal. If you suspect the Gateway(s) are not connecting, follow the
+  instructions in the [troubleshooting guide](/kb/administer/troubleshooting) to
+  diagnose the issue.
+</Alert>
+
+<Alert color="warning">
+  Azure disables ICMP for VMs without a public IP attached, so you won't be able
+  to ping internet hosts from the Firezone Gateway(s) or vice versa. This is
+  normal and expected. TCP and UDP traffic will work as expected using the
+  example configuration in this guide.
+</Alert>
+
+After a few minutes, you should see the Firezone Gateway(s) appear in the
+Firezone admin portal. You can now configure your Resources to use the new
+Firezone Gateway(s) you just provisioned.
+
+## Upgrading
+
+To upgrade the Firezone Gateway(s) to the latest version, simply run
+`terraform apply` again. This will trigger a redeployment of the Firezone
+Gateway(s) with the latest version by default.
```

Comment:
Have you tried this?  In a quick look it doesn't seem like there is anything that would be changing in the TF state, which would mean that nothing would get redeployed.  But, I may have overlooked something.

Reply:
Good catch. You're right. I've updated this guide to recommend setting the Gateway version explicitly.